lib/Cache/Redis.pm

```diff
@@ -125,10 +125,21 @@ sub get_multi {
 }
 
 sub set {
-    my ($self, $key, $value, $expire) = @_;
+    my ($self, $key, $value, $expire, $callback) = @_;
 
-    $self->_set($key, $value, $expire);
-    $self->{redis}->wait_all_responses unless $self->{nowait};
+    die 'set() requires key and value arguments' unless $key && $value;
+
+    if ($self->{nowait} && (!$callback || ($callback && ref $callback ne 'CODE')))
```

Comment:
This is too strict behaviour and potentialy breaks existing application.
Even if `nowait` with no callbacks, `set` value is stored later, so callback is not 
always required. 

Cache::Redis is cache module, so in some case developers can accord delay of cache reflection.